### DISCUSSION TITLE
MacOS native fullscreen mode.

### DISCUSSION THREAD

**Discussion by SalomonSmeke** [wezterm/wezterm#2808] [Q&A] [answered]
MacOS supports full-screen applications and provides them some OS support WRT gestures.

Westerm's full-screen mode appears to be a window-maximizing function, and macOS does not give the application a full screen "desktop" when enabled.

This disallows fast switching with "swipes" on a trackpad, and if I understand it correctly prevents "notch" handling on Macs with notches.

Relevant Apple dev docs: https://developer.apple.com/documentation/appkit/nsapplicationpresentationoptions/nsapplicationpresentationfullscreen
https://developer.apple.com/documentation/appkit/nswindow/1419527-togglefullscreen

"fullscreen" wezterm
![Screenshot 2022-11-28 at 09 00 10](https://user-images.githubusercontent.com/9869123/204310058-85d63700-0c03-460d-b650-446f787d9ece.png)

Fullscreen safari
![Screenshot 2022-11-28 at 09 00 20](https://user-images.githubusercontent.com/9869123/204310247-60231f47-67dc-491a-8ba8-548711b0dc7a.png)

Is this intentional? Would a ticket for native fullscreen support be welcome?

**Comment by gegoune** [comment, marked as answer]
Try `native_macos_fullscreen_mode = true`.

**Reply by SalomonSmeke** [reply]
:O how did I miss that!? thank you!

**Comment by SalomonSmeke** [comment]
Wow. `native_macos_fullscreen_mode = true`. Please close this. Thank you @gegoune <3
